kie-remote/kie-remote-services/src/main/java/org/kie/remote/services/rest/query/helpers/InternalProcInstQueryHelper.java

```diff
@@ -190,4 +185,43 @@ public static void badParameter(String paramName) {
        throw KieRemoteRestOperationException.badRequest("'" + paramName 
                + "' is not an accepted parameter for the rich process instance query operation" );
     }
+
+    private List<VariableInstanceLog> filterLastVarInstanceLogs(List<VariableInstanceLog> varLogs) {
+        // RHBPMS-4445 - doing this in memory rather than query
+        List<VariableInstanceLog> retVarLogs = new ArrayList<>();
+        Map<String, List<VariableInstanceLog>> varlogsMap = new HashMap<String, List<VariableInstanceLog>>();
+        for( VariableInstanceLog vlog : varLogs ) {
+            if( !varlogsMap.containsKey(vlog.getVariableId()) ) {
+                ArrayList<VariableInstanceLog> varlogsList = new ArrayList<VariableInstanceLog>();
+                varlogsList.add( vlog );
+                varlogsMap.put( vlog.getVariableId(), varlogsList );
+            } else {
+                varlogsMap.get( vlog.getVariableId() ).add( vlog );
+            }
+        }
+
+        // sort the vlogs by id (descending) and remove all but first
+        for( Map.Entry<String, List<VariableInstanceLog>> entry : varlogsMap.entrySet() ) {
+            List<VariableInstanceLog> value = entry.getValue();
+            Collections.sort(value, Collections.reverseOrder(new VariableInstanceLogIDComparator()));
+            entry.getValue().subList(1, entry.getValue().size()).clear();
```

Comment:
I guess it would make sense to check the size or catch IndexOutOfBoundException to not fail entire operation